Strings/2108/2108.c

```diff
@@ -0,0 +1,52 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+int main() {
+ 
+    char frase[101], palavra[101];
+    int maior, i;
+
+    maior = 0;    
+    do {
+        int lista_palavra[50], num_palavras;
+        scanf(" %101[^\n]", frase);
+
+        lista_palavra[0] = 0;
+        num_palavras = 1;
+        
+        if (frase[0] != '0') {
+            for(i=0;i<strlen(frase);i++) {
+                if (frase[i] == ' ') {
+                    lista_palavra[num_palavras] = i;
+                    num_palavras++;
+                }
+            }
+            lista_palavra[num_palavras] = strlen(frase);
+            
+            if (lista_palavra[1] - lista_palavra[0] >= maior) {
+                maior = lista_palavra[1] - lista_palavra[0];
+                strncpy(palavra, frase + 0, maior);
```

Comment:
```suggestion
                strncpy(palavra, frase + 0, maior);
                palavra[maior] = '\0';
```
Para corrigir o bug do uso da função `strncpy`, adiciona-se um caractere **\0** na última posição do vetor `palavra`, indicando o término da string quando for utilizada a função `printf(%s, palavra)`.